drivers/hmis/app/models/hmis/form/definition_validator.rb

```diff
@@ -42,6 +42,49 @@ def perform(document, role = nil, skip_cded_validation: false)
     @issues.errors
   end
 
+  # Method to validate individual CustomDataElementDefinition (CDED) against an associated form item.
+  # This is pulled out for reuse by the CustomDataElementGenerator
+  def self.validate_cded(item:, cded:)
```

Comment:
I made no changes to this logic, just pulled it out into a class method so the generator class could use it